kite-data/kite-data-hbase/src/main/java/org/kitesdk/data/hbase/avro/AvroEntitySerDe.java

```diff
@@ -331,20 +332,32 @@ private void initKACRecordDatumMaps(String fieldName, Schema fieldSchema,
     }
   }
 
+  private static Schema unwrapNull(Schema schema) {
+    if (schema.getType() == Schema.Type.UNION) {
+      List<Schema> types = schema.getTypes();
+      if (types.get(0).getType() == Schema.Type.NULL) {
+        return types.get(1);
+      } else {
+        return types.get(0);
+      }
+    }
+    return schema;
+  }
+
   private DatumReader<Object> buildDatumReader(Schema schema,
       Schema writtenSchema) {
     if (specific) {
-      return new SpecificDatumReader<Object>(writtenSchema, schema);
+      return new SpecificDatumReader<Object>(unwrapNull(writtenSchema), unwrapNull(schema));
```

Comment:
I think this is okay because we are only going to call this reader if there is a value. But I'm not sure whether we are changing assumptions for other parts of the code. If fields were assumed to be non-null in the past, we should make sure the code checks whether the reader can handle nulls and sets the correct default value.